public/app/features/alerting/unified/components/rules/Filter/RulesFilter.v2.tsx

```diff
@@ -1,226 +1,769 @@
+        </Stack>
+      </Stack>
+    </form>
+  );
+}
+
+interface FilterOptionsProps {
+  onSubmit: SubmitHandler<AdvancedFilters>;
+  onClear: () => void;
+  pluginsFilterEnabled: boolean;
+}
+
+const FilterOptions = ({ onSubmit, onClear, pluginsFilterEnabled }: FilterOptionsProps) => {
+  const styles = useStyles2(getStyles);
+  const theme = useStyles2((theme) => theme);
+  const { filterState } = useRulesFilter();
+  const isManualResetRef = useRef(false);
+
+  // Create portal container to render dropdowns above the popup modal
+  const portalContainer = usePortalContainer(theme.zIndex.portal + 100);
+
+  const defaultValues = searchQueryToDefaultValues(filterState);
+
+  // Fetch namespace and group data from all sources
+  const { currentData: grafanaPromRules = [], isLoading: isLoadingGrafanaPromRules } =
+    alertRuleApi.endpoints.prometheusRuleNamespaces.useQuery({
+      ruleSourceName: GRAFANA_RULES_SOURCE_NAME,
+    });
+
+  const { isLoading: isLoadingGrafanaRulerRules } = alertRuleApi.endpoints.rulerRules.useQuery({
+    rulerConfig: GRAFANA_RULER_CONFIG,
+  });
+
+  const externalDataSources = useMemo(getRulesDataSources, []);
+
+  const externalPromRulesQueries = externalDataSources.map((ds) =>
+    alertRuleApi.endpoints.prometheusRuleNamespaces.useQuery({
+      ruleSourceName: ds.name,
+    })
+  );
+
+  const isLoadingNamespaces = useMemo(() => {
+    return (
+      isLoadingGrafanaPromRules ||
+      isLoadingGrafanaRulerRules ||
+      externalPromRulesQueries.some((query) => query.isLoading)
     );
-  }, [activeTab, styles.content, styles.fixTabsMargin]);
+  }, [isLoadingGrafanaPromRules, isLoadingGrafanaRulerRules, externalPromRulesQueries]);
+
+  // Create namespace options with better display names and grouping
+  const namespaceOptions = useMemo((): Array<ComboboxOption<string>> => {
+    const grafanaFolders: Array<ComboboxOption<string>> = [];
+    const externalNamespaces: Array<ComboboxOption<string>> = [];
+
+    // Add Grafana folders
+    grafanaPromRules.forEach((namespace) => {
+      grafanaFolders.push({
+        label: namespace.name,
+        value: namespace.name,
+        description: t('alerting.rules-filter.grafana-folder', 'Grafana folder'),
+      });
+    });
+
+    // Add external data source namespaces
+    externalPromRulesQueries.forEach((query) => {
+      query.currentData?.forEach((namespace) => {
+        // Handle file paths from external Prometheus data sources
+        if (namespace.name.includes('/') && (namespace.name.endsWith('.yml') || namespace.name.endsWith('.yaml'))) {
+          const filename = namespace.name.split('/').pop() || namespace.name;
+          const maxDescriptionLength = 100;
+          const truncatedDescription =
+            namespace.name.length > maxDescriptionLength
+              ? `${namespace.name.substring(0, maxDescriptionLength)}...`
+              : namespace.name;
+
```

Comment:
How about extracting code responsible for providing autocomplete options to separate hooks?